core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java

```diff
@@ -231,7 +231,7 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
                 if (handshakeFailed &&
                     cause instanceof DecoderException &&
                     cause.getCause() instanceof SSLException) {
-                    // Swallow an SSLException raised after handshake failure.
+                    ctx.channel().attr(PENDING_EXCEPTION).set(cause);
```

Comment:
Last but not least - How about `new UnprocessedRequestException(cause.getCause())`?

Reply:
Q. Naively implementing this causes nested `UnprocessedRequestException`s due to the following line

https://github.com/line/armeria/blob/c260bfee7a68abbb5a6ca90814ed23d2e1f38f05/core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java#L458

Personally, I think it is ok to keep all exception wrapping logic at the `completeExceptionally()` layer in `HttpChannelPool`. 🤔  Let me know if I'm not understanding anything  correctly. 🙇

Reply:
That sounds good to me.